exchanges/gateio/ws_ob_resub_manager_test.go

```diff
@@ -0,0 +1,107 @@
+package gateio
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/thrasher-corp/gocryptotrader/common/key"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/subscription"
+	testexch "github.com/thrasher-corp/gocryptotrader/internal/testing/exchange"
+)
+
+func TestNewWSObResubManager(t *testing.T) {
+	t.Parallel()
+
+	m := newWSObResubManager()
+	require.NotNil(t, m)
+	assert.NotNil(t, m.lookup)
+}
+
+func TestIsResubscribing(t *testing.T) {
+	t.Parallel()
+
+	m := newWSObResubManager()
+	m.lookup[key.PairAsset{Base: currency.BTC.Item, Quote: currency.USDT.Item, Asset: asset.Spot}] = true
+	assert.True(t, m.IsResubscribing(currency.NewBTCUSDT(), asset.Spot))
```

Comment:
could be `require`

Reply:
Yeah will change once more nits come in. Thanks.